### ISSUE TITLE
Error: Uncaught (in promise): Error: No provider for Ng2DragDropService!

### ISSUE THREAD

**Issue by mateenQureshi** [ObaidUrRehman/ng2-drag-drop#80]
I am using ng2-drag-drop package in my angular 4 app, i have deployed to staging, the package works fine, but yesterday i deployed it live and started getting the error: 
**Error: Uncaught (in promise): Error: No provider for Ng2DragDropService!**

can you please help me to get out of this ASAP
Thanks,
Looking Forward

**Comment by ObaidUrRehman** [comment]
your prod build using AOT?

**Comment by mateenQureshi** [comment]
yes

**Comment by ObaidUrRehman** [comment]
No, let me look into this.

**Comment by dawnofthedev** [comment]
For us, after Angular upgrade we had to import in app module and call forRoot()

**Comment by ObaidUrRehman** [comment]
@mateenQureshi did calling `forRoot()` solve your problem?

**Comment by mateenQureshi** [comment]
No, actually without calling forRoot() fixes my issue during building, but when I navigate to the app, the component which is using ng2-drag-drop breaks the app by throwing the above exception (ref: title of the issue)

**Comment by mateenQureshi** [comment]
Let me know if u need any further details
Thank You

**Comment by seiyria** [comment]
I had this same problem (I was using 2.0.2, then noticed my most recent update broke when I pushed and it deployed). I upgraded to 2.9.2 and called .forRoot.
